### PR TITLE
Fix issue with threads named 'deleted-channel'

### LINKED ISSUE #1261: Thread names get exported as `#deleted-channel` in mentions
### Version

2.43.3

### Flavor

CLI (Command-Line Interface)

### Platform

Linux (Debian)

### Export format

HTML

### Steps to reproduce

- Make a mention to a thread in a channel
- Export that channel
- Observe the mention being converted to `#deleted-channel`

### Details

This piece of code is relevant;

https://github.com/Tyrrrz/DiscordChatExporter/blob/281e0f608ad0784bce1dc2d2cc7b76e419905ebd/DiscordChatExporter.Core/Exporting/HtmlMarkdownVisitor.cs#L284

It only looks for a name of the channel, not of a thread.

Even discord seems to struggle, not fetching a corresponding thread, only saying `#unknown`

### Checklist

- [X] I have looked through existing issues to make sure that this bug has not been reported before
- [X] I have provided a descriptive title for this issue
- [X] I have made sure that this bug is reproducible on the latest version of the application
- [X] I have provided all the information needed to reproduce this bug as efficiently as possible
- [ ] I have sponsored this project
- [ ] I have not read any of the above and just checked all the boxes to submit the issue

### PR DESCRIPTION
**Pull request by maxgsf**
Closes #1261

I tried to solve the problem with the thread labeled 'deleted-channel' by using populate threads before assigning the channel variable in the mention.

### CODE REVIEW

**Comment by Tyrrrz** [comment]
Thanks. This works, but pulling threads for an entire guild is notoriously slow.

**Comment by maxgsf** [comment]
Thanks for the feedback! I've updated my code to only fetch the necessary channel. Please check again.

File: DiscordChatExporter.Core/Exporting/ExportContext.cs [hunk 1]
```diff
@@ -48,6 +48,28 @@ var channel in Discord.GetGuildChannelsAsync(Request.Guild.Id, cancellationToken
...
+
+        if (_channelsById.ContainsKey(sid))
+        {
+            return;
+        }
+
+        var channel = await Discord.GetChannelAsync(sid, cancellationToken);
+
+        _channelsById[sid] = channel;
+    }
```
Comment (Tyrrrz) [inline DiscordChatExporter.Core/Exporting/ExportContext.cs:71]: Check for `null` at callsite instead

```suggestion
    public async ValueTask PopulateThreadAsync(
        Snowflake id,
        CancellationToken cancellationToken = default
    )
    {
        if (_channelsById.ContainsKey(id))
            return;

        var channel = await Discord.GetChannelAsync(id, cancellationToken);

        _channelsById[id] = channel;
    }
```

File: DiscordChatExporter.Core/Exporting/HtmlMarkdownVisitor.cs [hunk 2]
```diff
@@ -279,6 +279,8 @@ protected override async ValueTask VisitMentionAsync(
         }
         else if (mention.Kind == MentionKind.Channel)
         {
+            await context.PopulateThreadsAsync(mention.TargetId, cancellationToken);
```
Comment (Tyrrrz) [inline DiscordChatExporter.Core/Exporting/HtmlMarkdownVisitor.cs:282]:
```suggestion
        if (mention.TargetId is not null)
            await context.PopulateThreadsAsync(mention.TargetId, cancellationToken);
```

**Comment by Tyrrrz** [comment]
Looks like this PR has been abandoned

**Pull request closed without merging**